tests/schema/test_file.py

```diff
@@ -0,0 +1,55 @@
+import io
+
+import pytest
+
+from apistar import app, routing, schema, test, http, exceptions
+
+
+class File(schema.Object):
+    properties = {
+        'file': schema.File(max_length=15)
+    }
+
+
+def get_data(data: File) -> http.Response:
+    body = data['file'].read().decode('utf-8')
+    return http.Response({'body': body})
+
+
+app = app.App(routes=[
+    routing.Route('/upload/', 'POST', get_data),
+])
+
+client = test.TestClient(app)
+
+
+def test_read_data():
+    data = 'some buffer'
+    response = client.post('http://example.com/upload/', files={'file': io.StringIO(data)})
+    assert response.status_code == 200
+    assert response.json() == {'body': 'some buffer'}
+
+
+def test_not_file():
+    response = client.post('http://example.com/upload/', data={'file': 'some buffer'})
```

Comment:
No, this request will return the 200 response because the method `parse_form_data` returns the FileStorage object with stream `some buffer`